assets/js/components/ActivateAnalyticsCTA.js

```diff
@@ -44,11 +48,56 @@ export default function ActivateAnalyticsCTA( { children } ) {
 		select( CORE_MODULES ).isModuleActive( 'analytics' )
 	);
 
+	const analyticsModuleAvailable = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleAvailable( 'analytics' )
+	);
+	const [ inProgress, setInProgress ] = useState( false );
+
+	const isNavigatingToReauthURL = useSelect( ( select ) => {
+		if ( ! analyticsModuleAvailable ) {
+			return false;
+		}
+
+		const adminReauthURL = select( MODULES_ANALYTICS ).getAdminReauthURL();
+
+		if ( ! adminReauthURL ) {
+			return false;
+		}
+
+		return select( CORE_LOCATION ).isNavigatingTo( adminReauthURL );
+	} );
+
+	const isActivating = useSelect( ( select ) => {
+		if ( ! analyticsModuleAvailable ) {
+			return false;
+		}
+
+		return select( CORE_MODULES ).isFetchingSetModuleActivation(
+			'analytics',
+			true
+		);
+	} );
+
+	/*
+	 * Using debounce here because the spinner has to render across two separate calls.
+	 * Rather than risk it flickering on and off in between the reset call completing and
+	 * the navigate call starting, we will just set a debounce to keep the spinner for 3 seconds.
+	 */
+	const debouncedSetInProgress = useDebounce( setInProgress, 3000 );
+
+	useEffect( () => {
+		if ( isActivating || isNavigatingToReauthURL ) {
+			setInProgress( true );
+		} else {
+			debouncedSetInProgress( false );
+		}
+	}, [ isActivating, isNavigatingToReauthURL, debouncedSetInProgress ] );
```

Comment:
This part is more/less copied from [what we have implemented in `ResetButton`](https://github.com/google/site-kit-wp/blob/83b2edfd1bf05179172566acc8a9a4bb75fbeaf9/assets/js/components/ResetButton.js#L62-L75)

The idea here is that we want the in progress state to be set right away when the user clicks the button, but add a significant delay in setting it back to `false` due to the gap between activation and navigation in the case where Analytics needs to be activated before navigating to the setup.